src/main/java/org/disq_bio/disq/impl/file/Merger.java

```diff
@@ -0,0 +1,30 @@
+package org.disq_bio.disq.impl.file;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.hadoop.conf.Configuration;
+
+public class Merger {
+
+  private final FileSystemWrapper fileSystemWrapper;
+
+  public Merger() {
+    fileSystemWrapper = new HadoopFileSystemWrapper();
+  }
+
+  public void mergeParts(Configuration conf, String tempPartsDirectory, String outputFile)
+      throws IOException {
+    List<String> parts = fileSystemWrapper.listDirectory(conf, tempPartsDirectory);
+    List<String> filteredParts =
+        parts
+            .stream()
+            .filter(
+                f ->
+                    !(FilenameUtils.getBaseName(f).startsWith(".")
+                        || FilenameUtils.getBaseName(f).startsWith("_")))
+            .collect(Collectors.toList());
+    fileSystemWrapper.concat(conf, filteredParts, outputFile);
```

Comment:
As far as I know, `concat` fails on encrypted HDFS file systems.  We have a workaround in ADAM, mentioned previously.

Reply:
Ah good to know. There's a fallback in HadoopFileSystemWrapper.